streams/src/main/java/org/apache/kafka/streams/state/internals/WrappedStateStore.java

```diff
@@ -34,6 +34,9 @@
 public abstract class WrappedStateStore<S extends StateStore, K, V> implements StateStore, CachedStateStore<K, V> {
 
     public static boolean isTimestamped(final StateStore stateStore) {
+        if (stateStore instanceof KeyValueToTimestampedKeyValueByteStoreAdapter) {
```

Comment:
We should not add this case, but rather let `KeyValueToTimestampedKeyValueByteStoreAdapter` implement the marker interface `TimestampedBytesStore` such that the existing condition returns `true` for this store.

Reply:
but now` TimestampedBytesStore` like this:

```
public interface TimestampedBytesStore {
    static byte[] convertToTimestampedFormat(final byte[] plainValue) {
        if (plainValue == null) {
            return null;
        }
        return ByteBuffer
            .allocate(8 + plainValue.length)
            .putLong(NO_TIMESTAMP)
            .put(plainValue)
            .array();
    }
}
```

you mean we implement a method return true in this interface?